File: T7/Ejercicios Clase/Ejercicio 2 T6/src/model/Jefe.java
package model;

public class Jefe extends Trabajador{

    double acciones, beneficio;  // acciones --> porcentaje de acciones de la empresa


    public Jefe(){

    }

    public Jefe(String nombre, String apellido, String dni, double acciones, double beneficio) {
        super(nombre, apellido, dni);
        this.acciones = acciones;
        this.beneficio = beneficio;
    }


    @Override
    public void mostrarDatos() {
        super.mostrarDatos();
        System.out.println(" Acciones: " + acciones + " Beneficio: " + beneficio);
    }

    public double getAcciones() {
        return acciones;
    }

    public void setAcciones(double acciones) {
        this.acciones = acciones;
    }

    public double getBeneficio() {
        return beneficio;
    }

    public void setBeneficio(double beneficio) {
        this.beneficio = beneficio;
    }
}

File: Curso 2/PMPD/Tema1/TrabajadoresHerencia/src/model/Jefe.kt
package model

class Jefe(
    nombre: String,
    apellido: String,
    dni: String,
    salario: Double,
    var nResponsabilidad: Int
) : Trabajador(nombre, apellido, dni, salario),
    Sindicato {

    override fun calcularSalarioNeto(): Double {

        if (nResponsabilidad >= 3) {
            salario += (salario * 0.03)
            println("Te hemos subido el salario un 3%")
        } else {
            salario -= (salario * 0.03)
            println("Te hemos bajado el salario un 3%")
        }

        return salario
    }

    fun disminuirResponsabilidad() {
        if (nResponsabilidad > 0) {
            nResponsabilidad--
            println("Se ha reducido 1 de tu nivel responsabilidad")
        }
    }

    fun aumentarResponsabilidad() {
        if (nResponsabilidad < 5) {
            nResponsabilidad++
            println("Se ha aumentado 1 de tu nivel responsabilidad")
        }
    }


    override fun mostrarDatos() {
        super.mostrarDatos()
        println("nResponsabilidad: $nResponsabilidad")
    }

    override fun bajarSueldos(lista: ArrayList<Trabajador>): Boolean {
        println("Procedes a bajar los sueldos, eres jefe y puedes hacerlo")
        lista.forEach {
            if (it !is Jefe){
                it.salario *= 0.90
            }
        }
        return true
    }

    override fun calcularBeneficios(): Double {
        println("Como jefe, vas a calcular el beneficio de la empresa")
        return 0.0
    }

}